examples/integrating-with-ytt/internal-templating/go.mod

```diff
@@ -2,13 +2,11 @@ module example_internal_templating
 
 go 1.17
 
-require github.com/k14s/ytt v0.38.0
+replace github.com/vmware-tanzu/carvel-ytt => ../../../
```

Comment:
What motivates this `replace` directive?

As an example, I assume it's more useful when it's self-contained.